framework/deproxy_client.py

```diff
@@ -519,7 +525,7 @@ def handle_read(self):
                     self.h2_connection.increment_flow_control_window(
                         increment=event.flow_controlled_length, stream_id=None
                     )
-                    if event.stream_ended is None:
+                    if stream_is_not_ended:
```

Comment:
You can get stream states from H2Connection object:
```suggestion
                    if self.h2_connection._get_stream_by_id(event.stream_id).state_machine.state != h2.stream.StreamState.CLOSED:
```